src/gateway.c

```diff
@@ -207,81 +222,116 @@ static int rpc_server(void) {
 
     if (cfg->n_nodes == 1) {
         log_warn("No PEER NODE CONFIGURED. Terminating the RPC server...");
-        goto error_2;
+        goto error;
     }
-
-    for (i = 0; i < cfg->n_nodes - 1; i++) {
+    char client_addr[INET_ADDRSTRLEN];
```

Comment:
Where is "client_addr" used? Please remove "client_addr" if it is not used elsewhere.